io/src/lib.rs

```diff
@@ -0,0 +1,184 @@
+                Move::Paper | Move::Spock => true,
+            },
+            Move::Spock => match other {
+                Move::Paper | Move::Lizard | Move::Spock => false,
+                Move::Rock | Move::Scissors => true,
+            },
+        }
+    }
+}
+
+#[derive(Debug, Default, Encode, Decode, TypeInfo, Clone)]
+pub struct StageDescription {
+    pub anticipated_players: BTreeSet<ActorId>,
+    pub finished_players: BTreeSet<ActorId>,
+}
+
+#[derive(Debug, Default, Encode, Decode, TypeInfo, Clone)]
+pub enum GameStage {
+    #[default]
+    Preparation,
+    InProgress(StageDescription),
+    Reveal(StageDescription),
+}
+
+impl GameStage {
+    pub fn game_is_in_progress(&self) -> bool {
+        match self {
+            GameStage::Preparation => false,
+            GameStage::InProgress(_) | GameStage::Reveal(_) => true,
+        }
+    }
+
+    pub fn move_can_be_made(&self) -> bool {
+        match self {
+            GameStage::Preparation | GameStage::InProgress(_) => true,
+            GameStage::Reveal(_) => false,
+        }
+    }
+
+    pub fn is_player_in_game(&self, player: &ActorId) -> bool {
+        match self {
+            GameStage::Preparation => false,
+            GameStage::InProgress(description) => {
+                description.anticipated_players.contains(player)
+                    || description.finished_players.contains(player)
+            }
+            GameStage::Reveal(description) => {
+                description.anticipated_players.contains(player)
+                    || description.finished_players.contains(player)
+            }
+        }
+    }
+
+    pub fn current_players(&self) -> Option<BTreeSet<ActorId>> {
+        let description: &StageDescription;
+
+        match self {
+            GameStage::Preparation => return None,
+            GameStage::InProgress(progress_description) => {
+                description = progress_description;
+            }
+            GameStage::Reveal(reveal_description) => {
+                description = reveal_description;
+            }
+        }
+
+        let players = description
+            .anticipated_players
+            .union(&description.finished_players)
+            .copied()
+            .collect();
+        Some(players)
+    }
+}
+
+#[derive(Debug, Encode, Decode, TypeInfo)]
+pub struct Duration {
+    pub days: u64,
+    pub hours: u64,
+    pub minutes: u64,
+}
+
+#[derive(Debug, Encode, Decode, TypeInfo)]
+pub enum RevealResult {
+    Continue,
+    NextRoundStarted { players: BTreeSet<ActorId> },
+    GameOver { winner: ActorId },
+}
+
+#[derive(Debug, Decode, Encode, TypeInfo)]
+pub enum Action {
+    Register,
+    MakeMove(String),
+    Reveal(String),
+    ChangeNextGameConfig(GameConfig),
+    StopGame,
+}
+
+#[derive(Debug, Encode, Decode, TypeInfo)]
+pub enum Event {
+    PlayerRegistred,
+    SuccessfulMove(ActorId),
+    SuccessfulReveal(RevealResult),
+    GameConfigChanged,
+    GameWasStopped(BTreeSet<ActorId>),
```

Comment:
`Was` looks excessive.
```suggestion
    GameStopped(BTreeSet<ActorId>),
```